core/src/main/java/org/apache/brooklyn/util/core/internal/ssh/SshTool.java

```diff
@@ -49,7 +50,14 @@
      * These keys are detected from entity/global config and automatically applied to ssh executions. */
     public static final String BROOKLYN_CONFIG_KEY_PREFIX = "brooklyn.ssh.config.";
     
-    public static final ConfigKey<String> PROP_TOOL_CLASS = newStringConfigKey("tool.class", "SshTool implementation to use", null);
+    /**
+     * @deprecated since 0.9.0; use {@link SshMachineLocation#SSH_TOOL_CLASS}
+     */
+    @Deprecated
+    public static final ConfigKey<String> PROP_TOOL_CLASS = newStringConfigKey(
```

Comment:
We need a different key name for the class name of the WinRmTool and the SshTool. We'll use this by writing some configuration on the location (e.g. in brooklyn.properties on a named location), and will want to supply an implementation of WinRmTool and an implementation of SshTool - hence we need a different key name for each.

My reasoning for not having it in SshTool is that the config there is about config the SshTool understands. By the time you have an SshTool, then it has been instantiated. The place that instantiates the tool is SshMachineLocation, so that feels like a good place to define this config key. I also defined it in BrooklynConfigKeys, but with the common prefix "brooklyn.ssh.config.".